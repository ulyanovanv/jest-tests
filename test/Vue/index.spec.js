import { mount, createLocalVue, shallowMount} from '@vue/test-utils'
import index from '@/pages/index.vue'
import Vue from 'vue';
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import NuxtLink from '../../.nuxt/components/nuxt-link.client.js';

/*
https://github.com/vuetifyjs/vuetify/issues/4964
Currently not possible to use createLocalVue together with Vuetify
createLocalVue together with Vuetify creates multiple Vue instances
Solution: use Vue instead of createLocalVue
*/

// const localVue = createLocalVue()

describe('index', () => {
  Vue.use(Vuetify)
  Vue.use(Vuex)
  Vue.component('nuxt-link', NuxtLink)

  let getters, store

  beforeEach(() => {
    getters = {
      rates: () => {},
      currency_1: () => 'EUR',
      currency_2: () => 'USD',

      value_1: () => 1,
      value_2: () => 1.11
    }

    store = new Vuex.Store({
      modules: {
        getters
      }
    })
  })

  test('is a Vue instance', () => {
    // const wrapper = shallowMount(index, { store, localVue, stubs: ['nuxt-link'] })
    const wrapper = shallowMount(index, { store, Vue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('should render content correctly', () => {
    const wrapper = shallowMount(index, { store, Vue })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
