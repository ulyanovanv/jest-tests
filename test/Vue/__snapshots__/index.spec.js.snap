// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render content correctly 1`] = `
<v-layout-stub
  align="center"
  column=""
  justify="center"
  tag="div"
>
  <v-flex-stub
    md6=""
    sm8=""
    tag="div"
    xs12=""
  >
    <div
      class="text-center"
    >
      <h4>
        Welcome to Currency Converter!
      </h4>
       
      <nuxt-link-stub
        id="toConverterPage"
        prefetch="true"
        to="/converter"
      >
        You can try out our app!
      </nuxt-link-stub>
    </div>
  </v-flex-stub>
</v-layout-stub>
`;
